src/main/scala/org/apache/spark/ml/linalg/RAPIDSML.scala

```diff
@@ -73,6 +73,28 @@ private[spark] object RAPIDSML extends Serializable {
     jniRAPIDSML.dgemm(transa, transb, m, n, k, alpha, A.values, lda, B.values, ldb, beta, C.values, ldc, deviceID)
   }
 
+  /**
+   *
+   * @param transa int type representing the operation op(A) that is non- or (conj.) transpose.
+   * @param transb int type representing the operation op(B) that is non- or (conj.) transpose.
+   * @param m number of rows of matrix op(A) and C.
+   * @param n number of columns of matrix op(B) and C.
+   * @param k number of columns of op(A) and rows of op(B).
```

Comment:
Are these necessary as separate parameters, or can they be derived from other parameters?  For example, `A` is being passed as a `DenseMatrix`, can we simply query that object for its row count to get the `m` parameter?

Reply:
Yes, and the most original GEMM wrapper were only using DenseMatrix as input and query other parameters from it. 
But then we were thinking to align with RAFT APIs, so we proposed this PR: https://github.com/NVIDIA/spark-rapids-ml/pull/15, to make it more “familiar” for users who know GEMM well. 
(Actually from my feeling, GEMM parameters are really really confusing, at least for me at the beginning. during my implementation, I spent 20% time finding proper methods to construct column and 80% time thinking how the matrix should multiply with each other and filling proper parameters into this GEMM function. )


So, for GEMM users, I think "familar" parameters are better. For a freshman like me who has no gemm experience, I prefer the "less" parameters way. What do you think?

Reply:
We could supply both versions, one that is simplified with minimal parameters and one that looks like the full GEMM interface.